gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/TaskNameMatcher.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.tasks;
+
+import com.google.common.collect.Streams;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+// Copied from gradle-consistent-versions
```

Comment:
Note: I copied this from gradle-consistent-versions but did not copy over the test, hoping this is a temporary situation. It didn't seem like a good idea to add a full `implementation` dependency for the sake of this class, and I don't know that we currently have any Gradle utilities library for this to go into.

As an alternative: Since this plugin writes locks in a task action rather than during configuration time, maybe we can check the task graph directly instead of checking the start parameters. I haven't tried that yet.

Reply:
Looks like the task graph approach works.